packages/graphile-build-pg/src/plugins/PgIntrospectionPlugin.js

```diff
@@ -150,97 +150,102 @@ export default (async function PgIntrospectionPlugin(
+    //introspect the database and return memoized list of structures by kind (see the
+    //above types for the fields on each type)
+    const introspectionResultsByKind = await persistentMemoizeWithKey(cacheKey, () =>
+
+      //get a client to the postgres database
+      withPgClient(pgConfig, async pgClient => {
+
+        //load and run the query for the structure of the database
+        const introspectionQuery = await readFile(INTROSPECTION_PATH, "utf8");
+        const { rows } = await pgClient.query(introspectionQuery, [
+          schemas,
+          pgIncludeExtensionResources,
+        ]);
+
+        //sort the structures by kind
+        const pgStructuresByKind = rows.reduce(
+          (memo, { object }) => {
+            memo[object.kind].push(object);
+            return memo;
+          },
+          {
+            namespace: [],
+            class: [],
+            attribute: [],
+            type: [],
+            constraint: [],
+            procedure: [],
+            extension: [],
+          }
+        );
+
+        // Parse tags from comments (smart commenting system)
+        [
+          "namespace",
+          "class",
+          "attribute",
+          "type",
+          "constraint",
+          "procedure",
+          "extension",
+        ].forEach(kind => {
+          pgStructuresByKind[kind].forEach(object => {
+            if (pgEnableTags && object.description) {
+              const parsed = parseTags(object.description);
+              object.tags = parsed.tags;
+              object.description = parsed.text;
+            } else {
+              object.tags = {};
+            }
           });
+        });
 
-          for (const k in result) {
-            result[k].map(Object.freeze);
-          }
-          return Object.freeze(result);
-        })
-      )
+        const extensionConfigurationClassIds = flatMap(
+          pgStructuresByKind.extension,
+          e => e.configurationClassIds
+        );
+        pgStructuresByKind.class.forEach(klass => {
+          klass.isExtensionConfigurationTable =
+            extensionConfigurationClassIds.indexOf(klass.id) >= 0;
+        });
+
+        return pgStructuresByKind;
+      })
     );
 
+    //extract the schemas found in the database
     const knownSchemas = introspectionResultsByKind.namespace.map(n => n.name);
+
+    //check to see if any of the requested schemas are not actually present in the database
     const missingSchemas = schemas.filter(s => knownSchemas.indexOf(s) < 0);
     if (missingSchemas.length) {
-      const errorMessage = `You requested to use schema '${schemas.join(
-        "', '"
-      )}'; however we couldn't find some of those! Missing schemas are: '${missingSchemas.join(
-        "', '"
-      )}'`;
+      const errorMessage = `
+        You requested to use schema '${schemas.join("', '")}';
+        however we couldn't find some of those! Missing schemas are: '${missingSchemas.join("', '")}'
+      `;
```

Comment:
Please use `eslint --fix`; it formats the code for you (via prettier) to take code layout opinions out of the equation as much as possible. There are edge cases like this, but they are better addressed by extracting template expressions out to variables instead rather than manual formatting.

Reply:
Absolutely. The final commit in the PR will definitely have everything eslinted (and I will start doing it more in the intermediate ones as well). I was just moving some stuff around for my own visualizations.

Reply:
I recommend just having your editor do it for you every time you save the file. Its magic!